radix-engine-interface/src/hooks.rs

```diff
@@ -16,8 +17,21 @@ pub struct OnDropInput {}
 
 pub type OnDropOutput = ();
 
+// TODO: expose generic information, but fully-detailed actor?
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
-pub struct OnMoveInput {}
+pub struct OnMoveInput {
+    /// True if the node moves from caller to callee, otherwise false.
+    pub is_moving_down: bool,
+
+    /// True if the destination actor is a barrier, otherwise false.
+    pub is_to_barrier: bool,
+
+    /// True if the destination actor is auth zone, otherwise false.
+    pub is_to_auth_zone: bool,
```

Comment:
Couldn't this be removed and just use the `destination_blueprint_id`?

Reply:
Yes